src/lib.rs

```diff
@@ -218,17 +218,15 @@ pub async fn start_lemmy_server(args: CmdArgs) -> LemmyResult<()> {
   let mut interrupt = tokio::signal::unix::signal(SignalKind::interrupt())?;
   let mut terminate = tokio::signal::unix::signal(SignalKind::terminate())?;
 
-  if server.is_some() || federate.is_some() {
```

Comment:
Replace this with an environment variable check and set the variable in CI. This check can be removed in #4673.

Reply:
Most pipeline runners set CI env var automatically. Including Woodpecker you can use that one instead.

Reply:
That would work if there's an env var that's different for each step